pkg/queue/tenant_queues.go

```diff
@@ -142,10 +140,13 @@ func (q *tenantQueues) getOrAddQueue(tenant string, path []string, maxQueriers i
 		q.mapping.Put(tenant, uq)
 	}
 
-	if uq.maxQueriers != maxQueriers || uq.maxQueryCapacity != maxQueryCapacity {
-		uq.maxQueriers = maxQueriers
-		uq.maxQueryCapacity = maxQueryCapacity
-		uq.consumers = shuffleConsumersForTenants(uq.seed, uq.computeMaxQueriers(len(q.sortedConsumers)), q.sortedConsumers, nil)
+	consumersToSelect, err := q.limits.MaxConsumers(tenant, len(q.sortedConsumers))
+	if err != nil {
+		level.Error(util_log.Logger).Log("msg", "computing max consumers", "tenant", tenant, "error", err)
```

Comment:
Are we allowing a fallthrough here because `consumersToSelect` is expected to be set to `0` in the `MaxConsumers` function?

For the same reasons mentioned before, try not to rely on implementation outside this function. Set your expectations explicitly, or be very explicit about what the intended behaviour is. Assuming `consumersToSelect` is `0`, failing to compute the max consumers would result in... all queriers being allowable? That feels like a dangerous fallback.

Reply:
agree, this would allow all queriers.

the other option is to not create/return the queue which would cause the enqeue to fail. I thought that would be a bit extreme to do just because we are not able to fetch a limit.

But given this could only happen for an empty tenant and looking at the other comment, I think I'll go ahead with this.